test/task_ring_buffer_tests.cc

```diff
@@ -0,0 +1,50 @@
+#include "task_ring_buffer.h"
+
+#include <catch2/catch_test_macros.hpp>
+
+#include <celerity.h>
+
+#include "test_utils.h"
+
+namespace celerity::detail {
+
+TEST_CASE_METHOD(test_utils::runtime_fixture, "freeing task ring buffer capacity via horizons continues execution in runtime", "[task_ring_buffer]") {
+	using namespace std::chrono_literals;
+	celerity::distr_queue q;
+
+	std::atomic<bool> reached_ringbuffer_capacity = false;
+
+	auto observer = std::thread([&] {
+		while(runtime::get_instance().get_task_manager().get_total_task_count() < task_ringbuffer_size)
+			;
+		reached_ringbuffer_capacity = true;
+	});
+
+	celerity::buffer<int, 1> dependency{1};
+
+	for(size_t i = 0; i < task_ringbuffer_size + 10; ++i) {
+		q.submit(celerity::allow_by_ref, [=, &reached_ringbuffer_capacity](celerity::handler& cgh) {
+			celerity::accessor acc{dependency, cgh, celerity::access::all{}, celerity::read_write_host_task};
+			cgh.host_task(celerity::on_master_node, [=, &reached_ringbuffer_capacity] {
+				while(!reached_ringbuffer_capacity.load())
+					;
+			});
+		});
+	}
+
+	observer.join();
+}
+
+TEST_CASE_METHOD(test_utils::runtime_fixture, "deadlock in task ring buffer due to slot exhaustion is reported", "[task_ring_buffer]") {
+	celerity::distr_queue q;
+	CHECK_THROWS([&] {
```

Comment:
Consider using `CHECK_THROWS_WITH` so we're not accepting an unrelated exception type here.